java/ql/src/experimental/Security/CWE/CWE-522/InsecureBasicAuth.ql

```diff
@@ -66,25 +72,17 @@ class URLConstructor extends ClassInstanceExpr {
  * Gets a regular expression for matching private hosts.
  */
 private string getPrivateHostRegex() {
-  result = "localhost(/.*)?" or
-  result = "127\\.0\\.0\\.1(/.*)?" or // IPv4 patterns
-  result = "10(\\.[0-9]+){3}(/.*)?" or
-  result = "172\\.16(\\.[0-9]+){2}(/.*)?" or
-  result = "192.168(\\.[0-9]+){2}(/.*)?" or
-  result = "\\[0:0:0:0:0:0:0:1\\](/.*)?" or // IPv6 patterns
-  result = "\\[::1\\](/.*)?"
+  result = "(?i)localhost([:/].*)?|127\\.0\\.0\\.1([:/].*)?|10(\\.[0-9]+){3}([:/].*)?|172\\.16(\\.[0-9]+){2}([:/].*)?|192.168(\\.[0-9]+){2}([:/].*)?|\\[0:0:0:0:0:0:0:1\\]([:/].*)?|\\[::1\\]([:/].*)?" 
```

Comment:
It appears when using one of the URL/URI constructors with `host` parameter the square brackets around IPv6 addresses are optional:
`new URL("http", "[::1]", 8000, "")` is equivelent to `new URL("http", "::1", 8000, "")`

For a concatenated string that is not possible, but would likely cause a syntax exception so no special handling for this case by this query is needed I guess.
```suggestion
  result = "(?i)localhost([:/].*)?|127\\.0\\.0\\.1([:/].*)?|10(\\.[0-9]+){3}([:/].*)?|172\\.16(\\.[0-9]+){2}([:/].*)?|192.168(\\.[0-9]+){2}([:/].*)?|\\[?0:0:0:0:0:0:0:1\\]?([:/].*)?|\\[?::1\\]?([:/].*)?" 
```

Reply:
I've modified the regex as per the suggestion.